cfpb/consumerfinance.gov
# Change Featured Content body to plain text

The Featured Content Module (FCM) is documented [in the design system](https://cfpb.github.io/design-system/patterns/featured-content-module). Its Wagtail block implementation is [`v1.atomic_elements.organisms.FeaturedContent`](https://github.com/cfpb/consumerfinance.gov/blob/904c49d2db1492ea47b62771b318700754393ffc/cfgov/v1/atomic_elements/organisms.py#L1051).

The design documentation mentions "descriptive body copy text" and that

> Descriptions should be action-oriented and let the user know what they will get out of the featured content and how it’s related to the main page content, not just what it is. Descriptions should be 160 characters or less (including spaces).

This description and the documented examples suggest that the body text should be plain text and thus should not contain links, images, or text formatting). But in code this field is implemented as a Wagtail [`RichTextBlock`](https://docs.wagtail.io/en/stable/topics/streamfield.html#richtextblock), which allows for all of these things.

This commit modifies the FCM body field to instead be a [`CharBlock`](https://docs.wagtail.io/en/stable/topics/streamfield.html#charblock). This allows for only a single line of plain text.

## How to test this PR

Run a local server and create or edit a page with a Featured Content module. Note that the body is now plain text. Visit a live page with an FCM ([example](http://localhost:8000/data-research/research-reports/)) and notice that the body still renders correctly.

## Screenshots

Before -- FCM body could contain rich text elements including links and images:

![image](https://user-images.githubusercontent.com/654645/105074309-2b97fb80-5a56-11eb-98ce-1f1add58d39a.png)

After -- FCM body must be plain text:

![image](https://user-images.githubusercontent.com/654645/105074347-394d8100-5a56-11eb-82f9-aaad757c49a0.png)

## Notes and todos

See internal https://github.local/CFGOV/platform/issues/3808#issuecomment-274183 for some related discussion including a link to a detailed analysis of our existing production pages and how this won't affect them.

One design consideration: the Wagtail editing experience is now slightly annoying if you have a long sentence, as it cuts off a requires scrolling at ~100 characters. Our design guidance suggests 160 characters or less (and we exceed this often), so this is something users might run into often. On the other hand, maybe this is good because it encourages brevity?

Another option would be to use a [`TextBlock`](https://docs.wagtail.io/en/stable/topics/streamfield.html#textblock) here instead, which allows for multiple lines of input, and automatically wraps. That would look like this:

![image](https://user-images.githubusercontent.com/654645/105074757-cdb7e380-5a56-11eb-8905-4808c12473f3.png)

The one downside is that, because this supports newlines, it might mislead editors to think they can create multiple paragraphs, where in practice, they won't be able to. For example this input:

![image](https://user-images.githubusercontent.com/654645/105074893-fcce5500-5a56-11eb-99d1-160f9d10331c.png)

will still get rendered like this in the FCM:

![image](https://user-images.githubusercontent.com/654645/105074926-0b1c7100-5a57-11eb-9ec4-3088ff2ed143.png)

Any strong preferences on `CharBlock` vs `TextBlock`? @csebianlander or others?

## Checklist
